docs/site/Inside-Loopback-Application.md

```diff
@@ -0,0 +1,132 @@
+
+
+## Wiring Dependencies
+
+The `UserController` in the shopping example needs to connect to a user service to verify the user credentials and a jwt token service to create a token. Having these `dependencies` loosely coupled with the `UserController` will help developers of the `Login` usecase with separation of duties and inject mock services for rapid testing.
+
+> Dependency injection features in LoopBack is used to [wire dependencies](https://loopback.io/doc/en/lb4/Dependency-injection.html#flavors-of-dependency-injection) into constructors, class properties and methods.
+
+Dependency injection is used to wire services and repositories with controllers dynamically at run time. We will also see that this feature is extensively used across all LoopBack components. For example, datasources are injected into repository classes.
+
+## Models and Relations
+
+User and Order are domain objects in the shopping cart use case. They have the same structure and relations when posted by users from the front end as well as when persisted towards a backend datastore.
+
+- Loopback [Models](https://loopback.io/doc/en/lb4/Model.html#definition-of-a-model) define the structure of domain objects and can be used to represent any data store specific format including JSON.
+
+- [Model Relations](https://loopback.io/doc/en/lb4/Relations.html) help in defining entity composition and cardinalities like one-to-many or one-to-one.
+  - A user can have multiple orders, and hence the [User model](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/models/user.model.ts) has a has-many relation with the [Order model](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/models/order.model.ts)
+
+
+## Business logic
+
+- [Services](https://loopback.io/doc/en/lb4/Calling-other-APIs-and-web-services.html) act as the actual API logical layer and connect with repositories to fetch and create data. Services interchangeably, are also used to represent external services like other [apis, soap web-services and Rest APIs](https://loopback.io/doc/en/lb4/Calling-other-APIs-and-web-services.html
+). This comes handy in a microservices environment.
+
+- [Repositories](https://loopback.io/doc/en/lb4/Repositories.html) represent the entity layer for querying and persisting a domain model. They also connect with other repositories to resolve entity relations.
+  - To query and persist user data the example uses a [User repository](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/repositories/user.repository.ts). The 
+  user repository uses the entity composition defined in the `User model` to create a has-many relation with the [Order repository](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/repositories/order.repository.ts)
+
+
+## Connectors
```

Comment:
Connecting to databases

Reply:
agreed

Reply:
I have changed the title as "Entity layer and Database connectivity"